types/d3-scale/index.d.ts

```diff
@@ -1982,7 +1982,7 @@ export interface ScaleQuantize<Range, Unknown = never> {
  * @param range Array of range values.
  */
 // eslint-disable-next-line @definitelytyped/no-unnecessary-generics
-export function scaleQuantize<Range = number, Unknown = never>(range?: Iterable<Range>): ScaleQuantize<Range, Unknown>;
+export function scaleQuantize<Range = number | string, Unknown = never>(range?: Iterable<Range>): ScaleQuantize<Range, Unknown>;
```

Comment:
I'm not sure what this is fixing. The current types work with strings, the type that you changed is just a default if TypeScript can't infer the generic. Here is [your example](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wCYGY4DMoQhwDk2xA3ALABQNAxhAHarwMA20cAvJlgHSo6yNgFMAigFdkjGMABeIgBQBtAAwAaOAEYAupuUAiAEYEA7owOaDLESLZG2EkQZ0BKKtQbN4AN2FOtbjh2aEVVPgAWAE53OAB6OLhjMwsaBLgMuAA9AH56JhY4P0cRACYALjgWKGBGAHMgkKgwvgBWLVj06xhbexKDNMTM3JogA) which seems to type-check without issue.

Reply:
sorry my mistake

```
    const colorRange: string[] = [
      "#ebedf0",
      "#9be9a8",
      "#40c463",
      "#30a14e",
      "#216e39",
    ];
    const color = d3.scaleQuantize().domain([0, 4]).range(colorRange); //my mistake
    // const color = d3.scaleQuantize([0, 4], colorRange); // no error
```